fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceReferenceDAO.java

```diff
@@ -202,6 +202,13 @@ protected void insertResourceTokenRefs(String resourceType, Collection<ResourceT
                 } else {
                     ps.setNull(4, Types.INTEGER);
                 }
+
+                // compositeId can be null
+                if (xr.getCompositeId() != null) {
```

Comment:
seems very familiar to ParameterVisitorDao...

Reply:
yes/no. The processing in this DAO needs to handle upserts into the common_token_values table, which has to be handled differently for Derby, Db2 and PostgreSQL.